python/cudf/cudf/core/column/datetime.py

```diff
@@ -242,14 +242,20 @@ def normalize_binop_value(self, other: DatetimeLikeScalar) -> ScalarLike:
         if isinstance(other, (cudf.Scalar, ColumnBase, cudf.DateOffset)):
             return other
 
-        if isinstance(other, datetime.datetime):
-            other = np.datetime64(other)
-        elif isinstance(other, datetime.timedelta):
-            other = np.timedelta64(other)
-        elif isinstance(other, pd.Timestamp):
+        if isinstance(other, pd.Timestamp):
+            if other.tz is not None:
+                raise TypeError(
+                    "Cannot perform binary operation on timezone-naive columns"
+                    " and timezone-aware timestamps. For that, "
+                    "use `tz_localize`."
```

Comment:
```suggestion
                    "Cannot perform binary operation on timezone-naive columns"
                    " and timezone-aware timestamps. "
```

Does Pandas raise an error here? If not, should we raise a `NotImplementedError` instead?

Reply:
Ditto below; we should omit the part that says `"For that, use `tz_localize`" -- that is used in `tz_convert` as a hint to the user that "converting" from naive to tz-aware types needs the `tz_localize` function.

Reply:
Done 👍 Updated the below error messages too.